src/ipc/handler.c

```diff
@@ -720,16 +721,20 @@ static int ipc_dma_trace_config(uint32_t header)
 	uint32_t ring_size;
 #endif
 	struct sof_ipc_dma_trace_params_ext params;
+	uint64_t ticks;
 	int err;
 
 	/* copy message with ABI safe method */
 	IPC_COPY_CMD(params, _ipc->comp_data);
 
-	if (iCS(header) == SOF_IPC_TRACE_DMA_PARAMS_EXT)
-		platform_timer->delta = params.timestamp -
+	if (iCS(header) == SOF_IPC_TRACE_DMA_PARAMS_EXT) {
+		ticks = clock_ns_to_ticks(platform_ts->clk,
+					  params.timestamp_ns);
+		platform_timer->delta = ticks -
 					platform_timer_get(platform_timer);
```

Comment:
This calculation uses public platform timer API to get some information outside just to set another internal platform timer thing. Should be rather implemented as another public platform timer API that receives ts and performs the calculation (it might also simplify the new dependency on the platform default clock source index), so it could be called here like:
```c
platform_timer_set_delta(platform_timer, params.timetamp_ns)
```